engine/src/main/java/org/terasology/rendering/cameras/PerspectiveCamera.java

```diff
@@ -137,23 +138,25 @@ public void updateMatrices() {
     public void updateMatrices(float fov) {
         // Nothing to do...
         if (cachedPosition.equals(getPosition()) && cachedViewigDirection.equals(viewingDirection)
-                && cachedBobbingRotationOffsetFactor == bobbingRotationOffsetFactor && cachedBobbingVerticalOffsetFactor == bobbingVerticalOffsetFactor
-                && cachedFov == fov
-                && cachedZFar == getzFar() && cachedZNear == getzNear()
-                && cachedReflectionHeight == getReflectionHeight()) {
+            && cachedBobbingRotationOffsetFactor == bobbingRotationOffsetFactor && cachedBobbingVerticalOffsetFactor == bobbingVerticalOffsetFactor
+            && cachedFov == fov
+            && cachedZFar == getzFar() && cachedZNear == getzNear()
+            && cachedReflectionHeight == getReflectionHeight()) {
             return;
         }
 
         viewingDirection.cross(up, tempRightVector);
         tempRightVector.mul(bobbingRotationOffsetFactor);
 
-        projectionMatrix = createPerspectiveProjectionMatrix(fov, getzNear(), getzFar(),this.displayDevice);
+        float aspectRatio = (float) displayDevice.getDisplayWidth() / displayDevice.getDisplayHeight();
+        float fovY = (float) (2 * Math.atan2(Math.tan(0.5 * fov * TeraMath.DEG_TO_RAD), aspectRatio));
+        projectionMatrix.identity().perspective(fovY, aspectRatio, getzNear(), getzFar()).transpose();
```

Comment:
does this end up being the same as joml's [perspectiveRect](https://javadoc.io/static/org.joml/joml/1.9.25/org/joml/Matrix4fc.html#perspectiveRect-float-float-float-float-boolean-org.joml.Matrix4f-)? or is the fovY we come up with different than the perspective it would set if we passed display width and height to perspectiveRect?

either way, I think we can use the [`setPerspective`](https://javadoc.io/static/org.joml/joml/1.9.25/org/joml/Matrix4f.html#setPerspective-float-float-float-float-boolean-) would let us avoid the need to reset the matrix to `identity` first.

Reply:
also ... keeping the `.transpose` here but then re-transposing in the abstract `Camera.getProjectionMatrix`?

I guess the `projectionMatrix` property still needs to be transposed for some things, that's probably what you meant by things you need to correct in the camera later. But I lean toward adding an additional private field so we can store the non-transposed version instead of making a new object on every access.

Reply:
JOMLUtil already makes new instances when it goes back and forth so I don't see too much of an issue here. If the transpose goes away then the new instance just needs to be removed. keeping an extra field seems like it will introduce a bit more work when I go back to remove it.